aeron-driver/src/main/java/io/aeron/driver/PublicationParams.java

```diff
@@ -408,6 +411,34 @@ private void getUntetheredRestingTimeout(final ChannelUri channelUri, final Medi
             channelUri, UNTETHERED_RESTING_TIMEOUT_PARAM_NAME, ctx.untetheredRestingTimeoutNs());
     }
 
+    private void getMaxRetransmits(final ChannelUri channelUri)
+    {
+        final String maxRetransmtsString = channelUri.get(MAX_RETRANSMITS_PARAM_NAME);
+
+        if (maxRetransmtsString == null)
+        {
+            this.hasMaxRetransmits = false;
+
+            return;
+        }
+
+        try
+        {
+            maxRetransmits = Integer.parseInt(maxRetransmtsString);
+        }
+        catch (final NumberFormatException ex)
+        {
+            throw new IllegalArgumentException("invalid " + MAX_RETRANSMITS_PARAM_NAME + ", must be a number", ex);
+        }
+
+        if (maxRetransmits < 1)
+        {
+            throw new IllegalArgumentException("invalid " + MAX_RETRANSMITS_PARAM_NAME + ", must be > 0");
```

Comment:
I think we should have an upper bound on this as well because it will affect the amount of space allocated.